# LeviDNA
LeviDNA is an innovative, AI-driven bioinformatic pipeline designed for rapid and accurate eukaryotic environmental DNA (eDNA) classification and biodiversity assessment in deep-sea environments.
By fine-tuning the cutting-edge HyenaDNA architecture and integrating environmental metadata, Deep-Hyena minimizes reliance on incomplete reference databases, drastically reduces computational time, and enables the discovery and annotation of novel, divergent deep-sea species.
## The Problem
Deep-sea eDNA metabarcoding suffers from two critical limitations that hinder accurate biodiversity science:

- Poor Reference Database Representation: Traditional pipelines (QIIME2, DADA2, etc.) rely on sequence alignment against databases like SILVA or PR2. However, the majority of deep-sea eukaryotes are undiscovered or genetically divergent from shallow-water species. This results in numerous unassigned reads, leading to a severe underestimation of true biodiversity.

- Computational Bottleneck: Alignment-based methods scale poorly with the massive datasets generated by modern sequencing, requiring significant time and resources, which slows down urgent ecosystem monitoring.

 ## Current Progress
  HyenaDNA Finetunning on Downstream Classification on TATA
  - Dataset Link - [HuggingFace](https://huggingface.co/datasets/InstaDeepAI/nucleotide_transformer_downstream_tasks) This dataset was created for Nucleotide Transformers but suits well for our purpose.
    Evaluations
    
    <img width="647" height="216" alt="Screenshot from 2025-09-30 01-00-14" src="https://github.com/user-attachments/assets/61472511-643b-4581-a976-01ea0ac251a4" />

Confusion Matrix
<img width="930" height="989" alt="condusion_matrix" src="https://github.com/user-attachments/assets/bc00a939-a73e-4e43-aa56-a0ce46642631" />

## Installation Guide
# Clone the repository
```
git clone https://github.com/ayusrjn/LeviDNA.git
cd LeviDNA
```

# Create and activate environment
```
python -m venv env
source venv/bin/activate
```

# Install required packages
```
pip install -r requirements.txt
```
# Run the App
```
streamlit run main.py
```

## Our Approach
LeviDNA overcomes these challenges by replacing traditional alignment with a novel, sub-quadratic deep learning approach.

1. HyenaDNA for Sequence Classification
We fine-tune HyenaDNA, a genomic foundation model that uses the efficient Hyena Operator (implicit long convolutions) instead of the attention mechanism found in standard Transformers.
Novelty: HyenaDNA scales sub-quadratically with sequence length (O(L log L)), making the classification of long 18S or COI reads significantly faster and more memory-efficient than traditional O(L^2) Transformer models or O(L^2) alignment methods.

2. Alignment-Free: The model learns complex, long-range genomic signatures directly from the raw sequence (A, C, G, T), allowing it to accurately classify or cluster even highly divergent or novel deep-sea taxa that lack close matches in reference libraries.

3. Environmental Metadata Integration
To provide ecological context and improve clustering accuracy, we treat the analysis as a multi-modal learning problem.Environmental data (e.g., depth, temperature, pressure, location, sediment type) are used as auxiliary inputs to refine the sequence-based prediction. This helps anchor novel genetic clusters to specific deep-sea ecological niches (e.g., distinguishing between abyssal plain and hydrothermal vent-associated species).

